test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_interleave.cpp

```diff
@@ -0,0 +1,818 @@
+            mutable_command_buffer, nullptr, props, kernel, 1, nullptr,
+            &global_work_size, nullptr, 0, nullptr, nullptr, &command);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(mutable_command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, mutable_command_buffer, 0,
+                                          nullptr, nullptr);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        cl_mutable_dispatch_config_khr dispatch_config{
+            CL_STRUCTURE_TYPE_MUTABLE_DISPATCH_CONFIG_KHR,
+            nullptr,
+            command,
+            0 /* num_args */,
+            0 /* num_svm_arg */,
+            0 /* num_exec_infos */,
+            0 /* work_dim - 0 means no change to dimensions */,
+            nullptr /* arg_list */,
+            nullptr /* arg_svm_list - nullptr means no change*/,
+            nullptr /* exec_info_list */,
+            nullptr /* global_work_offset */,
+            nullptr /* global_work_size */,
+            nullptr /* local_work_size */
+        };
+        cl_mutable_base_config_khr mutable_config{
+            CL_STRUCTURE_TYPE_MUTABLE_BASE_CONFIG_KHR, nullptr, 1,
+            &dispatch_config
+        };
+
+        error =
+            clUpdateMutableCommandsKHR(mutable_command_buffer, &mutable_config);
+        test_error(error, "clUpdateMutableCommandsKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, mutable_command_buffer, 0,
+                                          nullptr, nullptr);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        // cl_ndrange_kernel_command_properties_khr test_props[] = { 0, 0, 0 };
+        size_t size;
+
+        cl_mutable_dispatch_exec_info_khr exec_info;
+
+        error = clGetMutableCommandInfoKHR(
+            command, CL_MUTABLE_DISPATCH_EXEC_INFO_KHR, sizeof(exec_info),
+            &exec_info, &size);
+        test_error(error, "clGetMutableCommandInfoKHR failed");
+
+        /*
+        if (test_local_work_size != local_size)
+        {
+            log_error("ERROR: Wrong size returned from "
+                      "clGetMutableCommandInfoKHR.");
+            return TEST_FAIL;
+        }
+        */
+
+        clSVMFree(context, pBuf);
+
+        return CL_SUCCESS;
+    }
+
+    size_t test_local_work_size = 0;
+    const size_t local_size = 3;
+    size_t size;
+
+    cl_mutable_command_khr command = nullptr;
+
+    clUpdateMutableCommandsKHR_fn clUpdateMutableCommandsKHR = nullptr;
+    clCommandQueueWrapper out_of_order_queue;
+    clCommandBufferWrapper mutable_command_buffer;
+
+    typedef struct
+    {
+        cl_int *pA;
+        cl_int *pB;
+        cl_int *pC;
+    } BufPtrs;
+};
+
+struct MutableUpdate : public BasicMutableCommandBufferTest
+{
+    using BasicMutableCommandBufferTest::BasicMutableCommandBufferTest;
```

Comment:
I don't know where to start with this particular test case:

**2.1.1 Multiple updates**
 Test using `clUpdateMutableCommandsKHR()` in a way which updates more than one command-handle in the same call.
 Test calling `clUpdateMutableCommandsKHR()` several times on the same command-handle and make sure that the most recent value persists.

Reply:
> Test using clUpdateMutableCommandsKHR() in a way which updates more than one command-handle in the same call.

The command-buffer will need more than one command, so we have more than one handle. Then for each of these handles there is a  `cl_mutable_dispatch_config_khr` entry in the `mutable_dispatch_list` member of `cl_mutable_base_config_khr`.

> Test calling clUpdateMutableCommandsKHR() several times on the same command-handle and make sure that the most recent value persists.

Call `clUpdateMutableCommandsKHR` with some new value in `cl_mutable_dispatch_config_khr`, then change the value in the same field and immediately call `clUpdateMutableCommandsKHR`  again. Then enqueue the command-buffer and verify that the most recent value set is the one which took effect,